src/main.cpp

```diff
@@ -2313,13 +2313,13 @@ bool DisconnectBlock(CBlock& block, CValidationState& state, CBlockIndex* pindex
     LogPrint("sc", "%s():%d - ===============> CBlockUndo red from DB:\n%s\n",
         __func__, __LINE__, blockUndo.ToString());
     // no coinbase in blockundo
-    if (blockUndo.vtxundo.size() < (block.vtx.size() - 1 + block.vcert.size()))
+    if (blockUndo.vtxundo.size() != (block.vtx.size() - 1 + block.vcert.size()))
         return error("DisconnectBlock(): block and undo data inconsistent");
 
-    for(int idx = block.vtx.size() - 1 + block.vcert.size(); idx < blockUndo.vtxundo.size(); idx++)
+    for(int idx = 0; idx < blockUndo.vVoidedCertUndo.size(); idx++)
```

Comment:
Are they to be processed in reverse order as in the loops below?

Reply:
Done